pkg/homedir/homedir.go

```diff
@@ -50,3 +38,47 @@ func GetCacheHome() (string, error) {
 	}
 	return filepath.Join(home, ".cache"), nil
 }
+
+// SetXdgDirs ensures the XDG_RUNTIME_DIR, DBUS_SESSION_BUS_ADDRESS and XDG_CONFIG_HOME env
+// variables are set when running as rootless.
+// containers/image uses XDG_RUNTIME_DIR to locate the auth file, XDG_CONFIG_HOME is
+// use for the containers.conf configuration file.
+func SetXdgDirs() error {
+	rootless := unshare.IsRootless()
+	if !rootless {
+		return nil
+	}
+
+	// Set up XDG_RUNTIME_DIR
+	runtimeDir := os.Getenv("XDG_RUNTIME_DIR")
+
+	if runtimeDir == "" {
+		var err error
+		runtimeDir, err = GetRuntimeDir()
+		if err != nil {
+			return err
+		}
+	}
+	if err := os.Setenv("XDG_RUNTIME_DIR", runtimeDir); err != nil {
+		return fmt.Errorf("cannot set XDG_RUNTIME_DIR: %w", err)
+	}
+
+	if rootless && os.Getenv("DBUS_SESSION_BUS_ADDRESS") == "" {
```

Comment:
There’s already an `if !rootless { return}` at the top.

Reply:
yeah, we don't need this, thanks for spotting it.

Not sure what change in the last version broke the Podman tests, I am still investigating